shifthelper/checks/__init__.py

```diff
@@ -115,15 +115,27 @@ def __init__(self,
     def run(self):
         while not self.stop_event.is_set():
             if len(self.queue) > 0:
-                if self.caller is not None:
-                    self.caller.place_call()
+
                 while len(self.queue) > 0:
                     message = self.queue.popleft()
                     self.logger.warning(message)
```

Comment:
I'm talking about both lines in conjunction. First you check the length of the queue. And then in the next line you pop it. In the meantime another thread might have emptied it. This applies only iff the  queue is modified by mutliple alert threads. I don't know if thats the case. Depending on the queue implementation (I don't know whether there are different kinds of queues in Python ) this might throw an exception when you try to pop an empty queue right? You're right. Maybe race condition is the wrong word. Anyways. All of that is not a reason to not merge this right now ;).